statsd/src/main/java/com/avast/metrics/statsd/StatsDGauge.java

```diff
@@ -0,0 +1,66 @@
+package com.avast.metrics.statsd;
+
+import com.avast.metrics.api.Gauge;
+import com.timgroup.statsd.StatsDClient;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.function.Supplier;
+
+public class StatsDGauge<T> implements Gauge<T> {
+    private final StatsDClient client;
+    private final String name;
+    private final Supplier<T> supplier;
+
+    StatsDGauge(final StatsDClient client, final String name, final Supplier<T> supplier) {
+        this.client = client;
+        this.name = name;
+        this.supplier = supplier;
+    }
+
+    @Override
+    public T getValue() {
+        return supplier.get();
+    }
+
+    void send() {
+        sendValue(getValue());
+    }
+
+    // copied from Dropwizard Graphite reporter
+    private void sendValue(Object o) {
```

Comment:
The StatsD client supports only `long` or `double` values, the value has to be converted to one of these.